eth/filters/api.go

```diff
@@ -327,15 +328,38 @@ func (api *PublicFilterAPI) NewFilter(crit FilterCriteria) (rpc.ID, error) {
 //
 // https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getlogs
 func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([]*types.Log, error) {
-	// Convert the RPC block numbers into internal representations
-	if crit.FromBlock == nil {
-		crit.FromBlock = big.NewInt(rpc.LatestBlockNumber.Int64())
+	if len(crit.Topics) > maxTopics {
+		return nil, errExceedMaxTopics
 	}
-	if crit.ToBlock == nil {
-		crit.ToBlock = big.NewInt(rpc.LatestBlockNumber.Int64())
+
+	var (
+		fromBlock int64
+		toBlock   int64
+	)
+
+	if crit.BlockHash != nil {
+		// look up block number from block hash
+		header, err := api.backend.HeaderByHash(ctx, *crit.BlockHash)
+		if err != nil {
+			return nil, err
+		}
+		if header == nil {
+			return nil, errors.New("unknown block")
+		}
+		fromBlock = int64(header.Number.Int64())
+		toBlock = fromBlock
+	} else {
+		// Convert the RPC block numbers into internal representations
+		if crit.FromBlock == nil {
+			fromBlock = int64(rpc.LatestBlockNumber)
+		}
+		if crit.ToBlock == nil {
+			toBlock = int64(rpc.LatestBlockNumber)
+		}
```

Comment:
If ToBlock != nil, should we assign toBlock = crit.ToBlock.Int64()? Same as fromBlock

Reply:
yes